### PR TITLE
Set up binder

### LINKED ISSUE #143: Enable binder on our repository
Binder 2.0 came out today. We should enable it on this repo so people can run the notebooks.

### PR DESCRIPTION
**Pull request by ixjlyons**
- [x] check that changing `user-environment.yml` -> `environment.yml` gets Binder to install dependencies
- [x] migrate all mention of `user-environment.yml` in docs, etc.
- [x] add badge to README
- [x] update `index.ipynb` to point to notebooks (rather than built HTML flies)
- [x] update build/deploy process to replace links in `index.ipynb`
- [x] see if it's possible to get Binder to enable the exercise2 extension

Fixes #143

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -15,6 +15,9 @@ Resonance: Learning Mechanical Vibration Engineering Through Computation
 .. image:: https://travis-ci.org/moorepants/resonance.svg?branch=master
    :target: https://travis-ci.org/moorepants/resonance
 
+.. image:: https://mybinder.org/badge.svg
+   :target: https://mybinder.org/v2/gh/moorepants/resonance/master
```
Comment (ixjlyons) [inline README.rst]: May want to change this to link directly to `index.ipynb` once it's usable on Binder.

**Comment by moorepants** [comment]
Cool, this is looking good. Thanks for tackling it!

**Comment by ixjlyons** [comment]
Ok I think this is pretty much done. The index notebook now has relative links to the other ipynb files so you can browse on GitHub and it works on Binder. I didn't put much effort into processing the index notebook because I think we'll eventually want to totally rework it or replace it with a web page with links to download the zip file, download the notebook, view the rendered HTML, launch the notebook on Binder etc.

**Comment by moorepants** [comment]
Sweet!

**Pull request merged**